.github/workflows/infracost.yml

```diff
@@ -0,0 +1,74 @@
+name: Infracost CI
+
+on:
+  pull_request:
+    paths:
+      - 'qhub/template/stages/02-infrastructure'
+      - '.github/workflows/infracost.yml'
+
+jobs:
+  infracost:
+    name: Run Infracost on QHub Infrastructure
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        provider:
+          - aws
+          - gcp
+          - azure
+
+    env:
+      TF_ROOT: qhub/template/stages/02-infrastructure/${{ matrix.provider }}
+
+    steps:
+      # Checkout the base branch of the pull request
+      - name: Checkout infrastructure
+        uses: actions/checkout@v3
+        with:
+          ref: '${{ github.event.pull_request.base.ref }}'
+
+      # Setup Infracost
+      - name: Setup Infracost
+        uses: infracost/actions/setup@v2
+        with:
+          api-key: ${{ secrets.INFRACOST_API_KEY }}
+
+      # Run Infracost on the base branch to generate the baseline cost
+      - name: Generate Infracost cost estimate baseline
```

Comment:
When I check the CI, I see there's [this warning](https://github.com/Quansight/qhub/runs/6793330134?check_suite_focus=true#step:4:16):
```
Warning: Input values were not provided for following Terraform variables: 'variable.name', 'variable.environment',
'variable.region', 'variable.kubernetes_version', 'variable.node_groups', 'variable.resource_group_name',
'variable.node_resource_group_name'. Use --terraform-var-file or --terraform-var to specify them.
```

Is this something we can/should handle now?